agent/tool-scripts/README.md

```diff
@@ -0,0 +1,219 @@
+# Tool Scripts (`agent/tool-scripts`)
+
+The "scripts" in this directory represent the "tools" that users can register
+to collect data from the systems related to a workload.  Each tool available
+for use with a `pbench-register-tool --name=<tool>` CLI invocation is listed
+in the `meta.json` file as either "transient" or "persistent".
+
+The Tool Meister running on each host where tools are registered is
+responsible for invoking that tool script to collect its generated data.  The
+commands to start and stop tools are sent to each Tool Meister at the
+appropriate time via the `pbench-init/start/stop/end/send/postprocess-tools`
+CLI interfaces.
+
+The tool scripts in this directory provide the interfaces used by the Tool
+Meister for checking for prerequisites, starting the tool data capture
+("datalog" scripts), stopping the tool data capture, and optionally
+post-processing the captured tool data ("postprocess" scripts).
+
+Most of the "tools" are symlinks to one script named `base-tool`.  This allows
+us to share all the common behaviors between the tools, while hanlding
+differences where appropriate.
+
+The `external-data-source`, `node-exporter`, and `dcgm` "tools" are handled
+separately since they operate a bit different from the rest.  See below for
+more details.
+
+## The Tool Life Cycle
+
+There are 4 phases, if you will, that the Tool Meister sub-system offers. The
+first phase is for executing all the "persistent" tools.  Once the Tool
+Meister sub-system is up and running, this phase is started first (required),
+and is ended before the Tool Meister sub-system is shut down (required).  The
+`pbench-init-tools` CLI interface begins the "persistent" tools phase, and the
+`pbench-end-tools` closes out the phase.  More about "persistent" tools below.
+
+The second phase is for executing all the "transient" tools (described below).
+This phase must be started after the "persistent" tools phase begins, and must
+end before the "persistent" tools phase ends.  However, the phase can repeated
+as many times as the caller likes.  The `pbench-start-tools` CLI begins the
+phase by start all the "transient" tools.  Then the caller must invoke the
+`pbench-stop-tools` CLI to shut down all the "transient" tools, ending this
+phase.
+
+The third phase is for all transient tools to send their data back to the host
```

Comment:
We should probably have a glossary and define terms like "pbench controller" (or whatever) and avoid circumlocutions in the rest of the documentation.

Reply:
Yes, that might help, but I am deliberately not using terms like that here because we want to have the tool phases independent of the rest of pbench.  The more I work on the Tool Meister interfaces, the more we should consider moving it to its own repo entirely.